mlxtend/plotting/corr_circle_pca.py

```diff
@@ -0,0 +1,97 @@
+# mlxtend Machine Learning Library Extensions
+#
+# A function for plotting a PCA correlation circle
+# File Author: Gabriel Azevedo Ferreira <az.fe.gabriel@gmail.com>
+
+from sklearn.decomposition import PCA
```

Comment:
there's also a PCA in mlxtend that I added a long time ago:

    from mlxtend.feature_extraction import PrincipalComponentAnalysis

They should provide equivalent results (both rely on SVD in scipy).

Reply:
I have just tried to replace it and the results seemed to be different. Maybe I need to tweak a little the output or provide a slightly different input. The code for each of the pictures is the same one I put in the docs in the Jupyter notebook (normalise the columns of iris and apply PCA). 

If it is ok, I will not change this for now and later try to figure out which one of the two is correct.

mlxtend  - PrincipalComponentAnalysis
![image](https://user-images.githubusercontent.com/19335547/58452890-a4dd2f00-814b-11e9-8455-5b2f86100518.png)

sklearn PCA
![Uploading image.png…]()

Reply:
It's a long time ago, but as far as I remember, the reason why I implemented my own PCA is that back then, scikit-learn didn't store the eigenvalues (or sth like that). Just checking, the mlxtend and scikit-learn impl. indeed produce the same results:

```python
from mlxtend.data import iris_data
from mlxtend.preprocessing import standardize
from mlxtend.feature_extraction import PrincipalComponentAnalysis
import numpy as np
import matplotlib.pyplot as plt

X, y = iris_data()
X = standardize(X)

pca = PrincipalComponentAnalysis(n_components=None)
pca.fit(X)
X_pca = pca.transform(X)

tot = sum(pca.e_vals_)
var_exp = [(i / tot)*100 for i in sorted(pca.e_vals_, reverse=True)]
cum_var_exp = np.cumsum(var_exp)

with plt.style.context('seaborn-whitegrid'):
    fig, ax = plt.subplots(figsize=(6, 4))
    plt.bar(range(4), var_exp, alpha=0.5, align='center',
            label='individual explained variance')
    plt.step(range(4), cum_var_exp, where='mid',
             label='cumulative explained variance')
    plt.ylabel('Explained variance ratio')
    plt.xlabel('Principal components')
    plt.xticks(range(4))
    ax.set_xticklabels(np.arange(1, X.shape[1] + 1))
    plt.legend(loc='best')
    plt.tight_layout()
```

![Unknown](https://user-images.githubusercontent.com/5618407/58483025-279ec380-8125-11e9-921e-4962d6b591c0.png)

and

```python
from sklearn.decomposition import PCA


pca = PCA(n_components=None)
pca.fit(X)
X_pca = pca.transform(X)

tot = sum(pca.explained_variance_)
var_exp = [(i / tot)*100 for i in sorted(pca.explained_variance_, reverse=True)]
cum_var_exp = np.cumsum(var_exp)

with plt.style.context('seaborn-whitegrid'):
    fig, ax = plt.subplots(figsize=(6, 4))
    plt.bar(range(4), var_exp, alpha=0.5, align='center',
            label='individual explained variance')
    plt.step(range(4), cum_var_exp, where='mid',
             label='cumulative explained variance')
    plt.ylabel('Explained variance ratio')
    plt.xlabel('Principal components')
    plt.xticks(range(4))
    ax.set_xticklabels(np.arange(1, X.shape[1] + 1))
    plt.legend(loc='best')
    plt.tight_layout()
```

![Unknown](https://user-images.githubusercontent.com/5618407/58483075-3c7b5700-8125-11e9-9f71-0ba0a3aaf9f1.png)


> Maybe I need to tweak a little the output or provide a slightly different input.

I think you are right though that the eigenvalues are not normalized in the mlxtend version  by default:

```python
pca = PrincipalComponentAnalysis(n_components=None)
pca.fit(X)
X_pca = pca.transform(X)
pca.e_vals_
```
```
array([2.91081808, 0.92122093, 0.14735328, 0.02060771])
```

(there was some reason why i wanted the raw eigenvalues, but I don't remember)

you could simply do that though via

```python
var_exp = [(i / tot)*100 for i in sorted(pca.explained_variance_, reverse=True)]
```

and see if it gives the same results then (I would be curious).

I can add

```python
var_exp = [(i / tot)*100 for i in sorted(pca.explained_variance_, reverse=True)]
```
also to the PCA class later as it might be useful in applications such as this (so that it doesn't have to be recomputed each time).

Btw should the arrow heads touch the circle or is the distance from the circle intended?

Reply:
I ran your code and it seems that 
    - When doing `standardize(X)`, the results from both mlxtend and sklearn are the same
    - When not  doing `standardize(X)`, the results are actually different
    - **When not doing `standardize(X)`, but removing the mean from the mlxtend (and from sklearn if desired), the results are the same again**
 
`from mlxtend.data import iris_data
from mlxtend.preprocessing import standardize
from mlxtend.feature_extraction import PrincipalComponentAnalysis
import numpy as np

X, y = iris_data()

X_no_mean = X - X.mean(axis=0)
X_standardized = standardize(X)

print("mlxtend")
for X_,name in zip([X,X_no_mean,X_standardized],['X','X_no_mean','X_standardized']):
    print('\t',name)
    pca = PrincipalComponentAnalysis(n_components=None)
    pca.fit(X_)
    X_pca = pca.transform(X_)
    print(X_pca[:4,:])  

print("sklearn")
for X_,name in zip([X,X_no_mean,X_standardized],['X','X_no_mean','X_standardized']):
    print('\t',name)
    pca = PCA()
    pca.fit(X_)
    X_pca = pca.transform(X_)
    print(X_pca[:4,:])`


About the heads touching the circle, we can thing that it touches the outer part of the n-th dimensional circle. So for the projection in a plane, it is always inside the circle. 

The sum of the squares of the correlations is 1, because of the ortogonality of the PCA transformation matrix.

Reply:
> When not doing standardize(X), but removing the mean from the mlxtend (and from sklearn if desired), the results are the same again

oh interesting. so they probably do automatic mean-centering in their code